bin/compile

```diff
@@ -1,102 +1,101 @@
 
-echo "-----> ⌛️  Setting env variables"
+  echo "-----> ✅  Role assumed successfully"
 
-# Set env variables for aws cli
-export AWS_ACCESS_KEY_ID="$ENV_AWS_ACCESS_KEY_ID"
-export AWS_SECRET_ACCESS_KEY="$ENV_AWS_SECRET_ACCESS_KEY"
+  echo "-----> ⌛️  Setting new temporary env variables from assumed role"
+  
+  # Unset env variables for aws cli
+  unset AWS_SECRET_ACCESS_KEY
+  unset AWS_ACCESS_KEY_ID
 
-echo "-----> ✅  Env variables set successfully"
+  # Set new temporary env variables for aws cli with permissions from assumed role
+  export AWS_ACCESS_KEY_ID="${STS[0]}"
+  export AWS_SECRET_ACCESS_KEY="${STS[1]}"
+  export AWS_SESSION_TOKEN="${STS[2]}"
 
+  echo "-----> ✅  New temp env variables set successfully"
 
-echo "-----> ⌛️  Assuming role"
+  echo "-----> ⌛️  Syncing bucket"
 
-# Assume role
-STS=($(aws sts assume-role --role-arn ${ENV_ROLE_ARN} --role-session-name "${ENV_HEROKU_APP_NAME}-${ENV_HEROKU_RELEASE_VERSION}" --duration-seconds 3600 --query 'Credentials.[AccessKeyId,SecretAccessKey,SessionToken]' --output text))
+  # Sync bucket
+  PUBLIC_FOLDER="$BUILD_DIR/public"
+  aws s3 sync "$PUBLIC_FOLDER" "s3://$ENV_S3_BUCKET_NAME" --quiet
 
-echo "-----> ✅  Role assumed successfully"
+  echo "-----> ✅  Synced bucket successfully"
 
+  echo "-----> ⌛️  Invalidating cache"
 
-echo "-----> ⌛️  Setting new temporary env variables from assumed role"
+  # Invalidate cache
+  # We need to invalidate the cache for manifest.json and manifest-assets.json
+  # So users get the last hash versioning
+  aws cloudfront create-invalidation --distribution-id "$ENV_DISTRIBUTION_ID" --paths "/vite/*manifest.json" "/vite/*manifest-assets.json" --no-cli-pager
 
-# Unset env variables for aws cli
-unset AWS_SECRET_ACCESS_KEY
-unset AWS_ACCESS_KEY_ID
-
-# Set new temporary env variables for aws cli with permissions from assumed role
-export AWS_ACCESS_KEY_ID="${STS[0]}"
-export AWS_SECRET_ACCESS_KEY="${STS[1]}"
-export AWS_SESSION_TOKEN="${STS[2]}"
-
-echo "-----> ✅  New temp env variables set successfully"
-
-
-echo "-----> ⌛️  Syncing bucket"
-
-# Sync bucket
-PUBLIC_FOLDER="$BUILD_DIR/public"
-aws s3 sync $PUBLIC_FOLDER s3://$ENV_S3_BUCKET_NAME --quiet
+  echo "-----> ✅  Invalidated cache successfully"
+else
+  echo "-----> ⏭️  Skipping AWS sync. Some required environment variables are not set:"
+  [ -z "$ENV_ROLE_ARN" ] && echo "ROLE_ARN is not set"
+  [ -z "$ENV_S3_BUCKET_NAME" ] && echo "S3_BUCKET_NAME is not set"
+  [ -z "$ENV_DISTRIBUTION_ID" ] && echo "DISTRIBUTION_ID is not set"
+  [ -z "$ENV_AWS_ACCESS_KEY_ID" ] && echo "AWS_ACCESS_KEY_ID is not set"
+  [ -z "$ENV_AWS_SECRET_ACCESS_KEY" ] && echo "AWS_SECRET_ACCESS_KEY is not set"
+  [ -z "$ENV_HEROKU_APP_NAME" ] && echo "HEROKU_APP_NAME is not set"
+  [ -z "$ENV_HEROKU_RELEASE_VERSION" ] && echo "HEROKU_RELEASE_VERSION is not set"
+fi
 
-echo "-----> ✅  Synced bucket successfully"
+echo "-----> ⌛️  Cleaning up slug"
 
+rm -rf "$BUILD_DIR/node_modules" \
```

Comment:
I think some of these could be included in `.slugignore` instead (plop, .vscode and .husky, maybe?). The buildpack should be responsible for cleaning only the things related to itself, that it consumed but are no longer needed. Makes sense?

[Heroku docs on .slugignore](https://devcenter.heroku.com/articles/slug-compiler#ignoring-files-with-slugignore)

Reply:
done 6b8bd8df6d06c970ffe9232fa1b54667101998e5